mlflow/data/http_dataset_source.py

```diff
@@ -0,0 +1,117 @@
+import os
+import posixpath
+import tempfile
+
+from typing import TypeVar, Any, Dict
+from urllib.parse import urlparse
+
+from mlflow.data.dataset_source import DatasetSource
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.utils.rest_utils import cloud_storage_http_request, augmented_raise_for_status
+
+
+HTTPDatasetSourceType = TypeVar("HTTPDatasetSourceType", bound="HTTPDatasetSource")
+
+
+class HTTPDatasetSource(DatasetSource):
+    """
+    Represents the source of a dataset stored at a web location and referred to
+    by an HTTP or HTTPS URL.
+    """
+
+    def __init__(self, url):
+        self._url = url
+
+    @property
+    def url(self):
+        """
+        The HTTP/S URL referring to the dataset source location.
+
+        :return: The HTTP/S URL referring to the dataset source filesystem location.
+        """
+        return self._url
+
+    @staticmethod
+    def _get_source_type() -> str:
+        return "http"
+
+    def load(self, dst_path=None) -> str:
+        """
+        Downloads the dataset source to the local filesystem.
+
+        :param dst_path: Path of the local filesystem destination directory to which to download the
+                         dataset source. If the directory does not exist, it is created. If
+                         unspecified, the dataset source is downloaded to a new uniquely-named
+                         directory on the local filesystem.
+        :return: The path to the downloaded dataset source on the local filesystem.
+        """
+        resp = cloud_storage_http_request(
+            method="GET",
+            url=self.url,
+        )
+        augmented_raise_for_status(resp)
```

Comment:
can we also add a test for a well formed URL that doesn't exist?

Reply:
Added a case for this to `test_source_load`